ReSharper.FSharp/test/data/features/daemon/typeHints/Patterns 02 - Match clauses.fs.gold

```diff
@@ -0,0 +1,82 @@
+﻿module Test
+
+let _ = function x||(0) -> ()
+
+match [Some 5] with
+| []||(1) -> ()
+| [x]||(2) -> ()
+| [x: int option]||(3) -> ()
+| [x; y]||(4) -> ()
+| [x; y: int option]||(5) -> ()
+| [_; x]||(6) -> ()
+| [_; _]||(7) -> ()
+| [x; _; _]||(8) -> ()
+| [x; _; Some 5]||(9) -> ()
+| x :: tail||(10) -> ()
+| _ :: (tail: int option list) -> ()
+| x: int option :: tail||(11) -> ()
+| x: int option :: _||(12) -> ()
+| x :: (tail: int option list) -> ()
+| ((x :: tail): int option list) -> ()
+| _ :: _||(13) -> ()
+| x :: _||(14) -> ()
+| _ :: x :: _||(15) -> ()
+| x :: _ :: _||(16) -> ()
+| x :: Some y||(17) :: _||(18) -> ()
```

Comment:
Do you think it makes sense to do it recursively here?

Reply:
I think for such cases I would like to see annotations for inner patterns as well

![image](https://github.com/user-attachments/assets/2245e951-3358-424e-85e0-54cb110ed6fb)

Reply:
All the inferred types are going to be shown in the outer hint anyway, though. I'm not sure if it's helpful to show it for non-generic fields types. While it might help somewhat to see the types next to the patterns, I think about how to prevent such hints from becoming too noisy.